apps/veyfi/utils/index.ts

```diff
@@ -16,3 +17,39 @@ export function getVotingPower(lockAmount: BigNumber, unlockTime: TMilliseconds)
 	}
 	return lockAmount.div(MAX_LOCK).mul(duration);
 }
+
+export const keyBy = <T1, T2 extends keyof T1 & string>(array: T1[], key: T2): TDict<T1 | undefined> => 
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	(array || []).reduce((r, x): TDict<T1> => ({...r, [(x as any)[key]]: x}), {});
+
+export const sort = <T>(data: T[], by: Extract<keyof T, string>, order?: 'asc' | 'desc'): T[] => {
+	const compare = (a: T, b: T): number => {
+		const elementA = a[by];
+		const elementB = b[by];
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		if (!isNaN(elementA as any) && !isNaN(elementB as any)) {
+			return order === 'desc' ? Number(elementA) - Number(elementB) : Number(elementB) - Number(elementA);
+		}
+		if (typeof elementA === 'string' && typeof elementB === 'string') {
```

Comment:
We could have an identical type guard here but for the strings, such as https://github.com/yearn/yearn.fi/pull/56/files?diff=unified&w=0#r1115618194